### PR TITLE
seccomp-tests.py: add distribution specific test cases

### PR DESCRIPTION
**Pull request by nasastry**
Add distribution specific test cases.
With the support of yaml, can run distribution and upstream test cases.

Signed-off-by: Nageswara R Sastry <rnsastry@linux.ibm.com>

### CODE REVIEW

**Comment by nasastry** [comment]
```
RHEL:
JOB ID     : 0a513f4d1c70ca635946b07728ccad64daa204af
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-09-27T03.33-0a513f4/job.log
 (1/2) seccomp-tests.py:SecComp.test;run-run_type-upstream-395e:  PASS (160.16 s)
 (2/2) seccomp-tests.py:SecComp.test;run-run_type-distro-94ba:  PASS (116.38 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-09-27T03.33-0a513f4/results.html
JOB TIME   : 287.64 s

SLES:
JOB ID     : d190872a38f1d7e05ff7814ab4e6599e6d490c3f
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-09-27T14.01-d190872/job.log
 (1/2) seccomp-tests.py:SecComp.test;run-run_type-upstream-395e:  PASS (128.12 s)
 (2/2) seccomp-tests.py:SecComp.test;run-run_type-distro-94ba:  PASS (346.87 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-09-27T14.01-d190872/results.html
JOB TIME   : 493.79 s
```

**Comment by nasastry** [comment]
```
SLES:
JOB ID     : 22e4c756cf761741166373292714f7e23027ff82
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-10-07T14.09-22e4c75/job.log
 (1/2) seccomp-tests.py:SecComp.test;run-run_type-upstream-395e: PASS (137.83 s)
 (2/2) seccomp-tests.py:SecComp.test;run-run_type-distro-94ba: PASS (391.18 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-10-07T14.09-22e4c75/results.html
JOB TIME   : 550.98 s

RHEL:
JOB ID     : a1fcb2535f7004e032093f26c477c6bb1968e664
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-10-07T03.42-a1fcb25/job.log
 (1/2) seccomp-tests.py:SecComp.test;run-run_type-upstream-395e: PASS (164.20 s)
 (2/2) seccomp-tests.py:SecComp.test;run-run_type-distro-94ba: PASS (122.68 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-10-07T03.42-a1fcb25/results.html
JOB TIME   : 299.09 s

```

File: security/seccomp-tests.py [hunk 1]
```diff
@@ -41,21 +41,38 @@ def setUp(self):
...
             if not smm.check_installed(package) and not smm.install(package):
                 self.cancel('%s is needed for the test to be run' % package)
-        url = "https://github.com/seccomp/libseccomp/archive/refs/heads/main.zip"
-        tarball = self.fetch_asset(url, expire='7d')
-        archive.extract(tarball, self.workdir)
-        self.sourcedir = os.path.join(self.workdir, 'libseccomp-main')
-        os.chdir(self.sourcedir)
-        process.run("./autogen.sh")
-        process.run("./configure")
+        run_type = self.params.get('type', default='upstream')
```
Comment (Naresh-ibm) [inline security/seccomp-tests.py:44]: @nasastry as discussed, change default='distro'

File: security/seccomp-tests.py [hunk 2]
```diff
@@ -41,21 +41,38 @@ def setUp(self):
...
+        if run_type == "upstream":
+            def_url = ("https://github.com/seccomp/libseccomp/archive/refs/"
+                       "heads/main.zip")
+            url = self.params.get('url', default=def_url)
```
Comment (Naresh-ibm) [inline security/seccomp-tests.py:48]: remove the default_url variable and make `default=None`

**Review by Naresh-ibm — CHANGES_REQUESTED** [review]

**Review by Naresh-ibm — APPROVED** [review]
@nasastry thanks for the PR
@abdhaleegit thanks for Reviews
LGTM

**Pull request merged**